src/UI/Renderer.php

```diff
@@ -20,4 +22,19 @@
 	 * @return string
 	 */
 	public function render($component);
+
+	/**
+	 * Get a new renderer with an additional context.
+	 *
+	 * A context makes it possible to use another renderer for (some) components when
+	 * they are renderer as subcomponents of a certain components. The use case that
+	 * spawned this functionality is the observation, that e.g. items representing
+	 * repository objects are renderer in different lists, where the individual items
+	 * look different every time but are morally the same item. Another use case could
+	 * be a kiosk mode for replaying tests or using ILIAS a LTI tool provider.
+	 *
+	 * @param  Component	$context
+	 * @return Renderer
+	 */
+	public function withAdditionalContext(Component $context);
```

Comment:
Asking for "Component" adds the requirement of all possible contexts needing a respective component. If you like to define a completely new layout for customer Amstutz, you would need to add a Context Amstutz, which implies, that we need a component Amstutz. Do we need a more flexible way? Maybe we can offer two methods here, one accepting any string and one asking for a component.